index.d.ts

```diff
@@ -0,0 +1,9 @@
+declare class SemanticReleaseError extends Error {
+    public name: string;
+    public message: string;
+    public code?: string;
+    public details?: string;
```

Comment:
Members are `public` by default so you don't have to specify that, unless you feel it enhances readability.

Reply:
I much prefer readability, but I'm happy for you to decide, I gather you're helping add typings to semantic-release. Let me know which you prefer!

Reply:
I really don't mind. S'up to you 😄Usually, if I don't have to, I don't do it, so leave the `public` off. However, if someone else has decided they want `public` specified, I'll do it to keep the consistency.